model/event/version_event.go

```diff
@@ -0,0 +1,42 @@
+package event
+
+import (
+	"time"
+
+	"github.com/mongodb/grip"
+)
+
+func init() {
+	registry.AddType(ResourceTypeVersion, versionEventDataFactory)
+	registry.AllowSubscription(ResourceTypeVersion, VersionStateChange)
+}
+
+func versionEventDataFactory() interface{} {
+	return &VersionEventData{}
+}
+
+const (
+	ResourceTypeVersion = "VERSION"
+	VersionStateChange  = "STATE_CHANGE"
+)
+
+type VersionEventData struct {
+	Status string `bson:"status,omitempty" json:"status,omitempty"`
+}
+
+func LogVersionStateChangeEvent(id, newStatus string) {
+	event := EventLogEntry{
+		Timestamp:    time.Now().Truncate(0).Round(time.Millisecond),
+		ResourceId:   id,
+		ResourceType: ResourceTypeVersion,
+		EventType:    VersionStateChange,
+		Data: &VersionEventData{
+			Status: newStatus,
+		},
+	}
+
+	logger := NewDBEventLogger(AllLogCollection)
+	if err := logger.LogEvent(&event); err != nil {
+		grip.Errorf("Error logging version event: %+v", err)
```

Comment:
use a structured message here.

Reply:
done

Reply:
@bsamek I've added EVG-3392 to add structure to all logging. Can you change the structure of this to match what i described there:
```
message.WrapError(err, message.Fields{
"resource_type": ..., 
"source": "event-log-fail", 
} 
```